pepagent/pepagent.py

```diff
@@ -66,222 +74,367 @@ def close_connection(self) -> None:
-                if isinstance(registry_paths, str):
-                    registry_paths = [registry_paths]
-
-                # check for valid registry paths
-                for rpath in registry_paths:
-                    if not is_valid_resgistry_path(rpath):
-                        # should we raise an error or just warn with the logger?
-                        raise ValueError(f"Invalid registry path supplied: '{rpath}'")
-
-                # dynamically build filter for set of registry paths
-                parametrized_filter = ""
-                for i in range(len(registry_paths)):
-                    parametrized_filter += "(namespace=%s and name=%s)"
-                    if i < len(registry_paths) - 1:
-                        parametrized_filter += " or "
-
-            sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where {parametrized_filter}"
-            flattened_registries = tuple(
-                chain(
-                    *[
-                        [r["namespace"], r["item"]]
-                        for r in map(
-                            lambda rpath: ubiquerg.parse_registry_path(rpath),
-                            registry_paths,
-                        )
-                    ]
-                )
-            )
-            results = self.run_sql_fetchall(sql_q, *flattened_registries)
+                sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where namespace = %s"
+                results = self.run_sql_fetchall(sql_q, namespace)
+
+        # Case 4. Get projects by namespace
+        elif tag:
+            sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where tag = %s"
+            results = self.run_sql_fetchall(sql_q, tag)
+            print(results)
 
-        # Case 3. Get projects by namespace
         else:
-            sql_q = f"select {NAME_COL}, {PROJ_COL} from {DB_TABLE_NAME} where namespace = %s"
-            results = self.run_sql_fetchall(sql_q, namespace)
+            _LOGGER.warning(f"Incorrect input!")
+            results = []
+
+        # extract out the project config dictionary from the query
+        return [peppy.Project().from_dict(p[1]) for p in results]
+
+    def get_projects_in_list(
+        self,
+        registry_paths: list,
+    ) -> List[peppy.Project]:
+        """
+        Get a list of projects as peppy.Project instances.
+        Get a list of projects in a list of registry_paths
+        :
+        :return: a list of peppy.Project instances for the requested projects.
+        """
+        if not isinstance(registry_paths, list):
+            raise TypeError(f"incorrect variable type provided")
+        for rpath in registry_paths:
+            if not is_valid_resgistry_path(rpath):
+                # should we raise an error or just warn with the logger?
+                raise ValueError(f"Invalid registry path supplied: '{rpath}'")
+
+        # dynamically build filter for set of registry paths
```

Comment:
@rafalstepien would probably call this a "code smell" and ask us to off load this to a helper function 😄

Reply:
you wrote this code, so I didn't change it LOL

Reply:
Yeahhhhhhh. I know.